packages/@react-stately/layout/src/TableLayout.ts

```diff
@@ -504,6 +518,53 @@ export class TableLayout<T> extends ListLayout<T> {
         }
         break;
       }
+      case 'section': {
+        // Add the section header to the visible layout infos since we always need to render it for labelling.
+        res.push(node.header);
+
+        // Note: the first/last row index calculated here is with respect to the section, not the index of the row in the collection itself
+        let firstVisibleRow = this.binarySearch(node.children, rect.topLeft, 'y');
+        let lastVisibleRow = this.binarySearch(node.children, rect.bottomRight, 'y');
+
+        // Add persisted rows before the visible rows.
+        let persistedRowIndices = this.persistedIndices.get(node.layoutInfo.key);
+        let persistIndex = 0;
+
+        while (
+          persistedRowIndices &&
+          persistIndex < persistedRowIndices.length &&
+          persistedRowIndices[persistIndex] < firstVisibleRow
+        ) {
+          let idx = persistedRowIndices[persistIndex];
+          if (idx < node.children.length) {
+            res.push(node.children[idx].layoutInfo);
+            this.addVisibleLayoutInfos(res, node.children[idx], rect);
+          }
+          persistIndex++;
+        }
+
+        for (let i = firstVisibleRow; i <= lastVisibleRow; i++) {
+          // Skip persisted rows that overlap with visible cells.
+          while (persistedRowIndices && persistIndex < persistedRowIndices.length && persistedRowIndices[persistIndex] < i) {
+            persistIndex++;
+          }
+
+          res.push(node.children[i].layoutInfo);
+          this.addVisibleLayoutInfos(res, node.children[i], rect);
+        }
+
+        // Add persisted rows after the visible rows.
```

Comment:
I'm not seeing a focused tablerow persisted when I scroll it out of view

Reply:
What story? I'm seeing the focused row + section persist when scrolled out of view:
<img width="1255" alt="image" src="https://user-images.githubusercontent.com/9661127/225696711-8a7a14f4-b764-446a-9dc6-9e37c0179e18.png">
Focused row is the `aria-rowindex=3` with the section row header above it. The next row in the dom is `aria-rowindex=6` which is non-contiguous since that was the next row that isn't persisted since it is close enough to being in view. https://reactspectrum.blob.core.windows.net/reactspectrum/0c9a33167b9201eea8842469e0d613fed710bb12/storybook/index.html?path=/story/tableview--dynamic-sections&providerSwitcher-express=false

Reply:
Ah, that story, but focus row 8 and scroll it off the bottom.

Reply:
Ah good catch, lemme fix